source/modulo_core/include/modulo_core/communication/MessagePair.hpp

```diff
@@ -52,6 +64,15 @@ class MessagePair : public MessagePairInterface {
    */
   void set_data(const std::shared_ptr<DataT>& data);
 
+protected:
```

Comment:
I need to add docstrings here, but I'll first wait to see if this new logic is ok with you.